app/services/conversation-channel.js

```diff
@@ -0,0 +1,38 @@
+import PhoenixSocket from 'phoenix/services/phoenix-socket';
+import { get } from '@ember/object';
+import { inject as service } from '@ember/service';
+
+export default PhoenixSocket.extend({
+  session: service(),
+  store: service(),
+
+  connect() {
+    if (!get(this, 'socket')) { // Only create one socket at a time
+      let token = get(this, 'session.session.authenticated.token');
+      this._super('ws://localhost:4000/socket', {
+        params: { token }
+      });
+    }
+  },
+
+  async join(conversation) {
+    await this.connect();
+    let id = get(conversation, 'id');
+    let channel = this.joinChannel(`conversation:${id}`);
+
+    channel.on('new:message', () => this._onNewMessage(conversation));
+  },
+
+  leave(conversation) {
+    let socket = get(this, 'socket');
+    let id = get(conversation, 'id');
+    let channel = socket.channel(`conversation:${id}`);
+    channel.leave();
+  },
+
+  _onNewMessage(conversation) {
+    let store = get(this, 'store');
+    let id = get(conversation, 'id');
+    store.findRecord('conversation', id);
```

Comment:
Could we not send the json payload for the conversation part with the event and then simply do `store.pushPayload`? 

This would save us two queries at least.

Reply:
We cannot do that because the user who sent the message would have the payload pushed twice, and ember-data would rightly reject it.

There also may be data that has changed on the backend since the payload was sent, so best to simply update the parent model, IMO.